arduino/waveform/Waveform.h

```diff
@@ -0,0 +1,125 @@
+/** @file Waveform.h
+  These functions can be included once interrupt based SPI transfer is available via Cyclops.
+    @code{.cpp}
+
+ void Waveform::prepare(){
+    if (cyclops->isAvailable()){
+        cyclops->dac_initiate_prog(source->nextVoltage()); // initiate will do chip selection. Non-blocking.
+        status = PREPARED;
+    }
+}
+  
+Waveform::latch(uint16_t delta){
+    if (state == PREPARED){
+        cyclops->dac_load();
+        state = LATCHED;
+        // time_rem was zero, this channel just got latched.
+        time_rem = source->holdTime() - delta;
+    }
+    source->stepForward(1);
+}
+    @endcode
+    @author Ananya Bahadur
+*/
+#ifndef CL_WAVEFORM_H
+#define CL_WAVEFORM_H
+
+#if ARDUINO >= 100
+ #include <Arduino.h>
+#else
+ #include <WProgram.h>
+#endif
+
+#include "Cyclops.h"
+#include "Source.h"
+
+/** @typedef waveformStatus */
+typedef enum {
+    INIT,
+    PREPARED,
+    LATCHED,
+    PAUSED
+} waveformStatus;
+
+
+/**
+ * @brief      Each Waveform Object has pointers to a Cyclops instance and a Source instance, effectively binding them together.
+ * @details    It can be in any one of 4 (waveformStatus) states: {``INIT``, 
+ *             ``PREPARED``, ``LATCHED``, ``PAUSED``} 
+ */
+class Waveform{
+ private:
+    uint16_t       time_rem;
+    waveformStatus backup_myStatus;
```

Comment:
When a waveform is paused it could be in `INIT`, `LATCHED`, `PREPARED` states, which is "backed up" and eventually restored when unpausing.